src/pretix/control/forms/waitinglist.py

```diff
@@ -45,12 +61,95 @@ def __init__(self, *args, **kwargs):
                 }
             )
             self.fields['subevent'].widget.choices = self.fields['subevent'].choices
+        else:
+            del self.fields['subevent']
+
+        if self.event.settings.waiting_list_names_asked:
+            self.fields['name_parts'] = NamePartsFormField(
+                max_length=255,
+                required=self.event.settings.waiting_list_names_required,
+                scheme=self.event.organizer.settings.name_scheme,
+                titles=self.event.organizer.settings.name_scheme_titles,
+                label=_('Name'),
+            )
+        else:
+            del self.fields['name_parts']
+
+        if not self.event.settings.waiting_list_names_asked:
+            del self.fields['phone']
+
+        choices = []
+
+        items = self.event.items.prefetch_related('variations').prefetch_related('quotas')
+        for item in items:
+            # don't offer items in the selection if they don't allow waitinglists or aren't on sale at all
+            if not item.allow_waitinglist | item.quotas.exists():
+                # except if they are currently set in the waitinglist, this will be then caught on submit in the clean step
+                if not self.instance.item.pk == item.pk:
+                    continue
+
+            if len(item.variations.all()) > 0:
+                for v in item.variations.all():
+                    choices.append((
+                        '{}-{}'.format(item.pk, v.pk),
+                        '{} – {}'.format(item, v.value) if item.active else mark_safe(
+                            f'<strike class="text-muted">{escape(item)} – {escape(v.value)}</strike>')
+                    ))
+            else:
+                choices.append(('{}'.format(item.pk), str(item) if item.active else mark_safe(
+                    f'<strike class="text-muted">{escape(item)}</strike>')))
+
+        self.fields['itemvar'].label = _("Product")
+        self.fields['itemvar'].help_text = _("Only includes Products with an enabled waiting list.")
+        self.fields['itemvar'].required = True
+        self.fields['itemvar'].widget = Select2ItemVarQuota(
+            attrs={
+                'data-model-select2': 'generic',
+                'data-select2-url': reverse('control:event.items.itemvars.select2', kwargs={
+                    'event': self.event.slug,
+                    'organizer': self.event.organizer.slug,
+                }),
+            },
+            choices=choices
+        )
+        self.fields['itemvar'].choices = choices
+
+    def clean(self):
+        if self.instance.voucher is not None:
+            self.add_error(None, _('A voucher for this waiting list entry was sent out already.'))
```

Comment:
You might want to raise a ValidationError here as once a voucher has been sent, there is no need to further validate.